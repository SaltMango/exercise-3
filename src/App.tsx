import React, { useState } from "react";
import "./app.css";
import CountryDetails from "./components/CountryDetails";
import ContinentCountries from "./components/ContinentCountries";

const App: React.FC = () => {
  const [selectedCountry, setSelectedCountry] = useState<string>("");
  const [continentCode, setContinentCode] = useState<string>("");

  return (
    <div className="container">
      <div className="county-style">
        <input
          className="input-style"
          value={selectedCountry}
          onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
            setSelectedCountry(e.target.value.toUpperCase())
          }
          type="text"
          placeholder="Enter Country Code"
        />
        <div className="details">
          <CountryDetails code={selectedCountry} />
        </div>
      </div>
      <div className="continent-style">
        <input
          className="input-style"
          type="text"
          value={continentCode}
          onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
            setContinentCode(e.target.value.toUpperCase())
          }
          placeholder="Enter continent Code"
        />
        <div className="details">
          {" "}
          <h4>Countries</h4>
          <ContinentCountries code={continentCode} />
        </div>
      </div>
    </div>
  );
};

export default App;
