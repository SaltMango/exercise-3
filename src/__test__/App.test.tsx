import React from "react";
import TestRenderer from "react-test-renderer";
// import { render, screen } from "@testing-library/react";
// import CountryDetails from "components/CountryDetails";
import App from "../App";

test("counter increment and decrement works correctly", () => {
  const tree = TestRenderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

// test("should CountryDetails component renders nothing initially ", () => {
//     render(<CountryDetails code={""}/>);
//     const countryDetails = screen.queryByTestId("countryDetails");
//     expect(countryDetails).

// });
