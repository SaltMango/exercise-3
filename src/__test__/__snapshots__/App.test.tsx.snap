// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`counter increment and decrement works correctly 1`] = `
<div
  className="container"
>
  <div
    className="county-style"
  >
    <input
      className="input-style"
      onChange={[Function]}
      placeholder="Enter Country Code"
      type="text"
      value=""
    />
    <div
      className="details"
    />
  </div>
  <div
    className="continent-style"
  >
    <input
      className="input-style"
      onChange={[Function]}
      placeholder="Enter continent Code"
      type="text"
      value=""
    />
    <div
      className="details"
    >
       
      <h4>
        Countries
      </h4>
    </div>
  </div>
</div>
`;
